core/storage/user/gae_models.py

```diff
@@ -2256,6 +2256,37 @@ def get_translation_reviewer_user_ids(cls, language_code):
             .fetch(keys_only=True))
         return [reviewer_key.id() for reviewer_key in reviewer_keys]
 
+    @classmethod
+    def get_number_of_reviewers_for_translations_in_lang_code(
```

Comment:
Do we actually want callers to use this, or is it only for auditing purposes? I ask because the contract now becomes slightly confusing: you can either retrieve the stats model, or query directly. Should we tighten the API & force callers to only use the stats model? Doing so would involve moving this & the other query code specifically to the audit job so that it can't be used for other purposes.

Reply:
Looks good--thanks!

Reply:
@BenHenning, Should we allow other layers to use the query method on the Model class directly? I think we have explicit classmethods for each query/filter as an endpoint. (This also allows storage layer codeowner to know (and validate) all the explicit methods provided by the storage layer.)

> Should we tighten the API & force callers to only use the stats model?

As we only have one such model I think it would be much simpler to just fetch the model and generate the count using the domain object. [Else we can have a classmethod `get_number_of_reviewers_for_translations_in_lang_code` instead of using query in other layers] (I think the first one is better)

wdyt?

Reply:
@DubeySandeep thanks for your thoughts. The auditing point is a good one, but isn't part of auditing models checking all uses of that model? I'd actually expect it to take the form of greping for the model name to see everywhere where it is used. Does this actually complicate auditing, and if so, how? That part isn't completely clear to me.

I'm not sure I follow your second point. The concern I raised earlier is that having a method in the storage model for computing an aggregation (e.g. by querying & counting the number of results) introduces a second way to retrieve that information: now it can be retrieved via storage and via the service layer. How should code authors pick one versus the other? A clean storage API means it's independent of the domain/service layer, and just provides the functionality it thinks is necessary. In this case, the extra confusion that comes from exposing that API feels like it's not worth exposing such functionality.

Note that this isn't a zero-cost alternative: the consequence assumption here is that directly querying models is already part of that model's public API. If that shouldn't be the case, then we need to change that directly (e.g. by forcing models to re-expose query functionality, or force them to hide it behind another function). I personally lean toward allowing query to be used outside of the model class, but this seems like something that's worth discussing further if folks have issue with that. If the outcome of that discussion is that we need to hide query behind a model class's functions, then the solution here would need to change.